core/tx_pool.go

```diff
@@ -1539,8 +1559,15 @@ func (pool *TxPool) demoteUnexecutables() {
 			pool.all.Remove(hash)
 			log.Trace("Removed old pending transaction", "hash", hash)
 		}
+		balance := pool.currentState.GetBalance(addr)
+		if !list.Empty() {
+			// Reduce the cost-cap by L1 rollup cost of the first tx if necessary. Other txs will get filtered out afterwards.
```

Comment:
I think this works, but it's a bit weird b/c there's no hard cap.

If there was a hard cap we could do something like enforce that the balance was enough to pay the cap & then place it into the tx priced list. Right now without the cap, it's not really clear when the tx would/could become valid again if it had low but present funds.